numpy/lib/arraypad.py

```diff
@@ -1208,7 +1208,7 @@ def pad(array, pad_width, mode, **kwargs):
     length to the vector argument with padded values replaced. It has the
     following signature::
 
-        padding_func(vector, iaxis_pad_width, iaxis, **kwargs)
```

Comment:
Hmm... I think the first is the signature (how the function is declared) while the fix shows  how it is called. This us one of those Python specific oddities, but I think the original is correct in that when writing the funtion it needs to be declared in that form.

Reply:
I'm not sure I understand what you're saying. Both the signature and the invocation are `func(..., kwargs)`, ie passing a dict as a positional argument. Note that passing expanded `kwargs` via `apply_along_axis` is relatively recent, which presumable explains why this passes them packed.

Reply:
@eric-wieser is correct the `kwargs` are simply passed to `apply_along_axis` as positional argument ([here](https://github.com/numpy/numpy/blob/master/numpy/lib/arraypad.py#L1354-L1359)). In fact having providing a function in the current form always raises a `TypeError` because `apply_along_axis` passes 4 _positional_ arguments to the function